specs/system_config.md

```diff
@@ -0,0 +1,71 @@
+  - [`batcherHash` (`bytes32`)](#batcherhash-bytes32)
+  - [`l1FeeOverhead` and `l1FeeScalar` (`uint256,uint256`)](#l1feeoverhead-and-l1feescalar-uint256uint256)
+- [Writing the system config](#writing-the-system-config)
+- [Reading the system config](#reading-the-system-config)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+The `SystemConfig` is a contract on L1 that can emit rollup configuration changes as log events.
+The rollup [block derivation process](./derivation.md) picks up on these log events, and applies the changes.
+
+## System config contents (version 0)
+
+Version 0 of the system configuration contract defines the following parameters:
+
+### `batcherHash` (`bytes32`)
+
+A versioned hash of the current authorized batcher sender(s), to rotate keys as batch-submitter.
+The first byte identifies the version.
+
+Version `0` embeds the current batch submitter ethereum address (`bytes20`) in the last 20 bytes of the versioned hash.
+
+In the future this versioned hash may become a commitment to a more extensive configuration,
+to enable more extensive redundancy and/or rotation configurations.
+
+### `l1FeeOverhead` and `l1FeeScalar` (`uint256,uint256`)
+
+The L1 fee parameters, also known as Gas Price Oracle (GPO) parameters,
+are updated in conjunction and apply new L1 costs to the L2 transactions.
+
+## Writing the system config
+
+The `SystemConfig` contract applies authentication to all writing contract functions,
+the configuration management can be configured to be any type of ethereum account or contract.
+
+On write, an event is emitted for the change to be picked up by the L2 system,
+and a copy of the new written configuration variable is retained in L1 state to read with L1 contracts.
+
+## Reading the system config
+
+A rollup node initializes its derivation process by finding a starting point based on its past L2 chain:
+
+- When started from L2 genesis, the initial system configuration is retrieved from the rollup chain configuration.
+- When started from an existing L2 chain, a previously included L1 block is determined as derivation starting point,
+  and the system config can thus be retrieved from the last L2 block that referenced the L1 block as L1 origin:
+  - `batcherHash`, `l1FeeOverhead` and `l1FeeScalar` are retrieved from the L1 block info transaction.
+  - other future variables may also be retrieved from other contents of the L2 block, such as the header.
+
+After preparing the initial system configuration for the given L1 starting input,
+with each new L1 block the system configuration is updated by processing all receipts from the block.
+
+The contained log events are filtered and processed as follows:
+
+- the log event contract address must match the rollup `SystemConfig` deployment
+- the first log event topic must match the ABI hash of `ConfigUpdate(uint256,uint8,bytes)`
+- the second topic determines the version. Unknown versions are critical derivation errors.
```

Comment:
Should we be rejecting invalid events instead of blowing up the op-node?

Reply:
Good point, ignoring the events is better, since a malicious/bad L1 system config contract upgrade can't harm the chain as much. Although when the system config is emitting bad things we might want to prefer safety over liveness. I'll open a ticket for this.

Reply:
Opened ENG-2974 and tackling next cycle. As long as we don't emit a bad event, handling it one way or another is compatible, so it does not block a testnet. But I would like to discuss more + update spec/implementation with the safety vs liveness decision.